File: 2099-find-subsequence-of-length-k-with-the-largest-sum/2099-find-subsequence-of-length-k-with-the-largest-sum.py
class Solution:
    def maxSubsequence(self, nums: List[int], k: int) -> List[int]:
        l = [list(t) for t in enumerate(nums)]
        l.sort(key=lambda x: x[1], reverse=True)
        l = l[0:k]
        l.sort(key=lambda x: x[0])
        return [el[1] for el in l]

File: 2099-find-subsequence-of-length-k-with-the-largest-sum/2099-find-subsequence-of-length-k-with-the-largest-sum.rb
# @param {Integer[]} nums
# @param {Integer} k
# @return {Integer[]}
def max_subsequence(nums, k)
    enum = nums.map.with_index { |n, i| [i, n] }
    enum.sort_by!{ |a| a[1] }.reverse!
    enum = enum[0...k]
    enum.sort_by!{ |a| a[0] }
    return enum.map{ |a| a[1] }
end